### ISSUE TITLE
rotating: Provide access to the rotated \textheight

### ISSUE THREAD

**Issue by schtandard** [latex3/latex2e#167]
(`latexbug` tells me `rotating` is third party software, but the documentation tells me that it is maintianed by the team (and it is contained in this repo), so I am opening the issue here. If that is wrong, feel free to point me into the right direction.)

## Brief outline of the issue
As can be seen from the MWE below, inside the `minipage` created by `rotating`'s `\@rotfloat` (which `sidewaysfigure` relies on) `\vsize` and `\textheight` still contain the values from outside the environment while `\hsize`, `\textwidth`, `\linewidth` and `\columnwidth` are (correctly) set to the outer value of `\textheight`. This means that one does not have access to the text height of the rotated page (i.e. the outer `\textwidth`) inside the `sidewaysfigure`, unless one manually stores the value before.

Naturally, one may want to scale images inside the `sidewaysfigure` using that length, so access to the value would be a useful addition to the command.

If changing `\vsize` and `\textheight` is not an option due to backwards compatibility, a new length, say `\rotatedtextheight` or `\outertextwidth`, would be a workable alternative.

## Minimal example showing the bug
```latex
\RequirePackage{latexbug}
\documentclass{article}

\usepackage{rotating}

\def\displaylengths{%
  \begin{tabular}{ l l }
    \dspllength\hsize
    \dspllength\vsize
    \dspllength\textwidth
    \dspllength\textheight
    \dspllength\linewidth
    \dspllength\columnwidth
  \end{tabular}%
}
\def\dspllength#1{%
  \texttt{\string#1} & \the#1 \\%
}

\begin{document}\makeatletter

\displaylengths

\begin{sidewaysfigure}
  \displaylengths
\end{sidewaysfigure}

\end{document}
```

## Log file
[testfile.log](https://github.com/latex3/latex2e/files/3505973/testfile.log)

**Comment by davidcarlisle** [comment]
Yes, I adopted rotating into the latex graphics bundle after Sebastian died.
We should update latexbug.

Will need to think about the rotating suggestion later.

**Comment by FrankMittelbach** [comment]
updated `latexbug`  ... the thinking still needs doing :-)

**Comment by FrankMittelbach** [comment]
@davidcarlisle for next release?

**Comment by davidcarlisle** [comment]
@FrankMittelbach  yes can do something...

**Comment by josephwright** [comment]
We'll look at the feature request but are not keen on extending the package interfaces
